### PR TITLE
maximum height from initial velocity

### PR DESCRIPTION
**Pull request by IrinaArtemova**
Add  maximum height from initial velocity to kinematic laws

### CODE REVIEW

File: maximum_height_from_velocity.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+def print_law() -> str:
+    return print_expression(law)
+
+
+@validate_input(initial_velocity_=initial_velocity)
+@validate_output(maximum_height)
+def calculate_maximum_height (initial_velocity_: Quantity) -> Quantity:
+    result_maximum_height = solve(law, maximum_height, dict=True)[0][maximum_height]
+    result_expr = result_maximum_heightsubs({initial_velocity: linitial_velocity_})
```
Comment (blackyblack) [inline maximum_height_from_velocity.py:35]: It won't work. Check your code before submission.

**Review by blackyblack — CHANGES_REQUESTED** [review]

File: maximum_height_from_velocity.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+"""Maximum_height_from_velocity.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
```
Comment (blackyblack) [inline maximum_height_from_velocity.py:6]: Remove unrelated comments.

File: maximum_height_from_velocity.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    https://colab.research.google.com/drive/1jWWV3KfbMQa5HFUbwPrCsceK0xoWuLcZ
+"""
+
+from sympy import (Eq, solve)
```
Comment (blackyblack) [inline maximum_height_from_velocity.py:10]: Laws without tests cannot be approved.

File: maximum_height_from_velocity.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+from symplyphysics import (units, Quantity, Symbol, print_expression, validate_input, validate_output)
+
+# Description
+## The maximum height to which a body thrown vertically upwards will rise depends on the initial velocity
+## Law: h = (v**2)/(2*g)
+## v is initial velocity,
+## g is acceleration of free fall.
+
+initial_velocity = Symbol("initial_velocity", units.velocity)
+free_fall_acceleration = Symbol("free_fall_acceleration", units.acceleration)
```
Comment (blackyblack) [inline maximum_height_from_velocity.py:20]: Free fall acceleration is already present in sympy. You can use it instead your own.

File: maximum_height_from_velocity.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+law = Eq(maximum_height, initial_velocity**2 /(2* free_fall_acceleration))
+
+def print_law() -> str:
+    return print_expression(law)
+
+@validate_input(initial_velocity_=initial_velocity)
+@validate_output(maximum_height)
+def calculate_maximum_height (initial_velocity_: Quantity) -> Quantity:
+    result_maximum_height = solve(law, maximum_height, dict=True)[0][maximum_height]
+    result_expr = result_maximum_height.subs({initial_velocity: initial_velocity_})
```
Comment (blackyblack) [inline maximum_height_from_velocity.py:31]: Free fall acceleration is a free symbol - no value attached to it. Therefore you won't get an actual quantity from this function.

**Comment by IrinaArtemova** [comment]
Removed unrelated comments
Added tests
Used free fall acceleration from sympy

File: maximum_height_from_velocity_test.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
+from collections import namedtuple
```
Comment (blackyblack) [inline maximum_height_from_velocity_test.py:1]: Tests are in `test` folder.

File: maximum_height_from_velocity_test.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    SI,
+)
+from symplyphysics.laws.kinematic import maximum_height_from_velocity as maximum_height_law
+
+# Description
+## If a body is thrown vertically upwards at a speed of 10 m/s, it will rise to a height of  (10*10) /2*9,8 = 5.1 meters
+
+@fixture(name="test_args")
+def test_args_fixture():
+     v = Quantity(10 * units.meter / units.second)
```
Comment (blackyblack) [inline maximum_height_from_velocity_test.py:17]:
```suggestion
    v = Quantity(10 * units.meter / units.second)
```

File: maximum_height_from_velocity_test.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    assert SI.get_dimension_system().equivalent_dims(result.dimension, units.length)
+    result_vector = convert_to(result, units.meter).evalf(2)
+    assert result_vector == approx(5.10, 0.01)
+
+def test_bad_velocity(test_args):
+    vb = Quantity(1 * units.coulomb)
+    with raises(errors.UnitsError):
+       maximum_height_law.calculate_height(vb)
+    with raises(TypeError):
+       maximum_height_law.calculate_height(100)
```
Comment (blackyblack) [inline maximum_height_from_velocity_test.py:32]: Add empty line.

File: symplyphysics/laws/kinematic/maximum_height_from_velocity.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+"""Maximum_height_from_velocity.ipynb
```
Comment (blackyblack) [inline symplyphysics/laws/kinematic/maximum_height_from_velocity.py:2]: Remove unrelated comments.

File: symplyphysics/laws/kinematic/maximum_height_from_velocity.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+# Description
+## The maximum height to which a body thrown vertically upwards will rise depends on the initial velocity
+## Law: h = (v**2)/(2*g)
+## v is initial velocity,
+## g is acceleration of free fall.
+
+maximum_height = Symbol("maximum_height", units.length)
+initial_velocity = Symbol("initial_velocity", units.velocity)
+
+law = Eq(maximum_height, initial_velocity**2 /(2*units.acceleration_due_to_gravity))
```
Comment (blackyblack) [inline symplyphysics/laws/kinematic/maximum_height_from_velocity.py:21]:
```suggestion
law = Eq(maximum_height, initial_velocity**2 / (2 * units.acceleration_due_to_gravity))
```

File: symplyphysics/laws/kinematic/maximum_height_from_velocity.py [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+def print_law() -> str:
+    return print_expression(law)
+
+
+@validate_input(initial_velocity_=initial_velocity)
+@validate_output(maximum_height)
+def calculate_maximum_height (initial_velocity_: Quantity) -> Quantity:
+    result_maximum_height = solve(law, maximum_height, dict=True)[0][maximum_height]
+    result_expr = result_maximum_height.subs({initial_velocity: initial_velocity_})
+    return Quantity(result_expr)
```
Comment (blackyblack) [inline symplyphysics/laws/kinematic/maximum_height_from_velocity.py:33]: Empty line.

**Comment by IrinaArtemova** [comment]
updated law and test, added test to test folder

File: maximum_height_from_velocity_test.py (hunk 6 above)
Comment (blackyblack) [inline maximum_height_from_velocity_test.py:1]: Delete duplicate file.

File: test/kinematic/maximum_height_from_velocity_test.py [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+    result_vector = convert_to(result, units.meter).evalf(2)
+    assert result_vector == approx(5.10, 0.01)
+
+
+def test_bad_velocity(test_args):
+    vb = Quantity(1 * units.coulomb)
+    with raises(errors.UnitsError):
+       maximum_height_law.calculate_height(vb)
+    with raises(TypeError):
+       maximum_height_law.calculate_height(100)
```
Comment (blackyblack) [inline test/kinematic/maximum_height_from_velocity_test.py:34]: Add empty line.

File: symplyphysics/laws/kinematic/maximum_height_from_velocity.py [hunk 13]
```diff
@@ -0,0 +1,25 @@
+from sympy import (Eq, solve)
```
Comment (blackyblack) [inline symplyphysics/laws/kinematic/maximum_height_from_velocity.py:1]: Move law and test to `dynamics` category.

File: maximum_height_from_velocity_test.py (hunk 6 above)
Comment (IrinaArtemova) [reply maximum_height_from_velocity_test.py]: done

File: test/kinematic/maximum_height_from_velocity_test.py (hunk 12 above)
Comment (IrinaArtemova) [reply test/kinematic/maximum_height_from_velocity_test.py]: added

File: symplyphysics/laws/kinematic/maximum_height_from_velocity.py (hunk 13 above)
Comment (IrinaArtemova) [reply symplyphysics/laws/kinematic/maximum_height_from_velocity.py]: moved

File: symplyphysics/laws/dynamics/maximum_height_from_velocity.py [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+law = Eq(maximum_height, initial_velocity**2 / (2 * units.acceleration_due_to_gravity))
+
+
+def print_law() -> str:
+    return print_expression(law)
+
+
+@validate_input(initial_velocity_=initial_velocity)
+@validate_output(maximum_height)
+
```
Comment (blackyblack) [inline symplyphysics/laws/dynamics/maximum_height_from_velocity.py:21]: Remove empty line

File: test/dynamics/maximum_height_from_velocity_test.py [hunk 15]
```diff
@@ -0,0 +1,36 @@
+from collections import namedtuple
+from pytest import approx, fixture, raises
+from symplyphysics import (
+    errors,
+    units,
+    convert_to,
+    Quantity,
+    SI,
+)
+from symplyphysics.laws.kinematic import maximum_height_from_velocity as maximum_height_law
```
Comment (blackyblack) [inline test/dynamics/maximum_height_from_velocity_test.py:10]: It won't work. Check your code before submission.

File: test/dynamics/maximum_height_from_velocity_test.py [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+
+
+def test_basic_height(test_args):
+    result = maximum_height_law.calculate_height(test_args.v)
+    assert SI.get_dimension_system().equivalent_dims(result.dimension, units.length)
+    result_vector = convert_to(result, units.meter).evalf(2)
+    assert result_vector == approx(5.10, 0.01)
+
+
+def test_bad_velocity(test_args):
```
Comment (blackyblack) [inline test/dynamics/maximum_height_from_velocity_test.py:29]: `test_args` is unused.

**Review by blackyblack — CHANGES_REQUESTED** [review]

File: test/dynamics/maximum_height_from_velocity_test.py (hunk 15 above)
Comment (IrinaArtemova) [reply test/dynamics/maximum_height_from_velocity_test.py]: changed law path

File: test/dynamics/maximum_height_from_velocity_test.py [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+
+@fixture(name="test_args")
+def test_args_fixture():
+    v = Quantity(10 * units.meter / units.second)
+    Args = namedtuple("Args", ["v"])
+    return Args(v=v)
+
+
+def test_basic_height(test_args):
+    result = maximum_height_law.calculate_height(test_args.v)
```
Comment (blackyblack) [inline test/dynamics/maximum_height_from_velocity_test.py:23]: It does not work. Check your code before submission.

**Review by blackyblack — CHANGES_REQUESTED** [review]

File: test/dynamics/maximum_height_from_velocity_test.py (hunk 17 above)
Comment (IrinaArtemova) [reply test/dynamics/maximum_height_from_velocity_test.py]: changed function name

**Review by blackyblack — APPROVED** [review]

**Pull request merged**